Add BibDesk Integration plugin

alberti42 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Your release `main.js` contains multiple long strings that are base64 encoded, which contain code, That is obfuscation, and is not allowed as per our developer policies.

[constructor(app:App,manifest:PluginManifest) {](https://github.com/alberti42/obsidian-bibdesk-integration/blob/345a165d2c96b9d6c3ee09490349e54d4131290c/src/main.ts#L24-L42)
This code should be in `onload`, not in the constructor.

[// Path to plugins folder](https://github.com/alberti42/obsidian-bibdesk-integration/blob/345a165d2c96b9d6c3ee09490349e54d4131290c/src/main.ts#L35-L39)
To get the folder of your plugin use `this.manifest.dir`

[atob("QGFydGljbGV7RWluc3RlaW46MTkzNSwKICAgIGF1dGhvciA9IHtFaW5zdGVpbiwgQS4gYW5kIFBvZG9sc2t5LCBCLiBhbmQgUm9zZW4sIE4ufSwKICAgIGRvaSA9IHsxMC4xMTAzL1BoeXNSZXYuNDcuNzc3fSwKICAgIGpvdXJuYWwgPSB7UGh5cy4gUmV2Ln0sCiAgICBtb250aCA9IHttYXl9LAogICAgbnVtYmVyID0gezEwfSwKICAgIHBhZ2VzID0gezc3N30sCiAgICB0aXRsZSA9IHt7Q2FuIFF1YW50dW0tTWVjaGFuaWNhbCBEZXNjcmlwdGlvbiBvZiBQaHlzaWNhbCBSZWFsaXR5IEJlIENvbnNpZGVyZWQgQ29tcGxldGU/fX0sCiAgICB2b2x1bWUgPSB7NDd9LAogICAgeWVhciA9IHsxOTM1fX0KCkBhcnRpY2xle1dhdHNvbjoxOTUzLAogICAgQXV0aG9yID0ge1dhdHNvbiwgSi4gRC4gYW5kIENyaWNrLCBGLiBILiBDLn0sCiAgICBUaXRsZSA9IHt7TW9sZWN1bGFyIFN0cnVjdHVyZSBvZiBOdWNsZWljIEFjaWRzOiBBIFN0cnVjdHVyZSBmb3IgRGVveHlyaWJvc2UgTnVjbGVpYyBBY2lkfX0sCiAgICBKb3VybmFsID0ge05hdHVyZX0sCiAgICBZZWFyID0gezE5NTN9LAogICAgVm9sdW1lID0gezE3MX0sCiAgICBOdW1iZXIgPSB7NDM1Nn0sCiAgICBQYWdlcyA9IHs3Mzd9LAogICAgTW9udGggPSB7YXByfSwKICAgIERvaSA9IHsxMC4xMDM4LzE3MTczN2EwfSwKfQoKQGFydGljbGV7RnJhbmtsaW46MTk1MywKICAgIEF1dGhvciA9IHtGcmFua2xpbiwgUm9zYWxpbmQgRS4gYW5kIEdvc2xpbmcsIFIuIEcufSwKICAgIFRpdGxlID0ge3tFdmlkZW5jZSBmb3IgMi1DaGFpbiBIZWxpeCBpbiBDcnlzdGFsbGluZSBTdHJ1Y3R1cmUgb2YgU29kaXVtIERlb3h5cmlib251Y2xlYXRlfX0sCiAgICBKb3VybmFsID0ge05hdHVyZX0sCiAgICBZZWFyID0gezE5NTN9LAogICAgVm9sdW1lID0gezE3Mn0sCiAgICBOdW1iZXIgPSB7NDM2OX0sCiAgICBQYWdlcyA9IHsxNTZ9LAogICAgTW9udGggPSB7anVsfSwKICAgIERvaSA9IHsxMC4xMDM4LzE3MjE1NmEwfSwKfQo=")](https://github.com/alberti42/obsidian-bibdesk-integration/blob/345a165d2c96b9d6c3ee09490349e54d4131290c/src/defaults.ts#L26)
Just include the actual content in the plugin as a string, having this base64 encoded looks very phishy.

[function max(a:number,b:number): number {](https://github.com/alberti42/obsidian-bibdesk-integration/blob/345a165d2c96b9d6c3ee09490349e54d4131290c/src/bibtex_parser.worker.ts#L11-L19)
Why are you reimplementing these instead of using `Math.max` and `Math,min` ?

alberti42 commented:
Thanks! I did all changes.

PS: Do you get notificiations when I post a message like this, or I need to tag you to inform you when changes are made?

alberti42 reviewed (commented):
Made requested changes to `bibdesk-integration`. New version 1.0.2 available.

alberti42 commented:
Dear Jothei, what does the flag "Installation not recommended" mean? Is the plugin rejected or some changes are needed?

alberti42 commented:
Ok. I think I understood why the flag `Installation not recommended` was added. Because of the obfuscated code which you pointed out above, right? OK. In the resubmission of the plugin, I avoided any base64 encoding, just pure text.

joethei commented:
Sorry for the delay.

[# Obsidian BibDesk Integration](https://github.com/alberti42/obsidian-bibdesk-integration/blob/cfecc85539f64cc24d832d5e369b11e0fb163b98/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

["description": "Allows using BibTex citations and opening attached PDFs from a BibDesk library.",](https://github.com/alberti42/obsidian-bibdesk-integration/blob/cfecc85539f64cc24d832d5e369b11e0fb163b98/manifest.json#L6)
Try to avoid starting your description with "Allows you to" or similar wording.
A lot of plugin descriptions already use that, it's getting a bit repetitive.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

alberti42 commented:
Thanks!

Yes, I changed the header in the README.md and updated the description. The new changes are included in [release 1.0.3](https://github.com/alberti42/obsidian-bibdesk-integration/releases/tag/1.0.3)

alberti42 commented:
@joethei I know you a lot of work and probably some backlog. Is there anything more has to be done on this plugin? It was nicely crafted and I am sure it will be helpful for people to manage their scientific library. It is many months since it was submitted. Thanks for your consideration.

alberti42 commented:
@joethei Is there anything I should do on my side or simply wait? Perhaps you are simply busy with other projects? A short answer would be greatly appreciated.

joethei commented:
You have not updated the description in the manifest.

alberti42 commented:
@joethei Thanks! I had changed `package.json` but forgot to also adjust `manifest.json` with the new description.

Now the new [release 1.0.4](https://github.com/alberti42/obsidian-bibdesk-integration/releases/tag/1.0.4) includes the new description.
